src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const stateArray = [];
+
+  for (const action of actions) {
+    transform(stateCopy, action);
+
+    const varState = { ...stateCopy };
+
+    stateArray.push(varState);
+  }
+
+  return stateArray;
+}
+
+function transform(state, action) {
+  switch (action.type) {
+    case 'addProperties':
```

Comment:
Consider will be better if you create a variables for these magic words and use them everywhere

Reply:
> Consider will be better if you create a variables for these magic words and use them everywhere

I got it. Fixed it

Reply:
not fixed 
The Previous mentor means that you need to create constant for addProperties, removeProperties, clear and reuse it, for instance:
```
const ACTIONS ={
 ADD,
 REMOVE,
 CLEAR
};
...
case ACTIONS.ADD:
```

Reply:
Ok, I realized what the mistake was. Attempt №3